app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package com.wire.android.ui.calling.incoming
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.navigation.BackStackMode
+import com.wire.android.navigation.EXTRA_CONVERSATION_ID
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.navigation.parseIntoQualifiedID
+import com.wire.kalium.logic.feature.call.usecase.RejectCallUseCase
+import com.wire.kalium.logic.feature.call.AnswerCallUseCase
+import com.wire.kalium.logic.feature.conversation.ObserveConversationDetailsUseCase
+import com.wire.kalium.logic.data.conversation.ConversationDetails
+import com.wire.kalium.logic.data.id.ConversationId
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class IncomingCallViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val navigationManager: NavigationManager,
+    private val conversationDetails: ObserveConversationDetailsUseCase,
+    private val rejectCall: RejectCallUseCase,
+    private val acceptCall: AnswerCallUseCase
+) : ViewModel() {
+    var callState by mutableStateOf(IncomingCallState())
+        private set
+
+    val conversationId: ConversationId? = savedStateHandle
+        .getLiveData<String>(EXTRA_CONVERSATION_ID)
+        .value
+        ?.parseIntoQualifiedID()
```

Comment:
We don't need to pass it as a liveData and I think the `conversationId` should be nullable as we don't have to display this screen when it's null

Reply:
Removed the `liveData` but had to keep it as nullable as `savedStateHandle`.`get(..)` returns a @Nullable

Reply:
Yes, we throw an exception in that case, we don't have to that screen